test/integration/targets/async/tasks/main.yml

```
@@ -163,8 +168,12 @@
     that:
     - async_result.ansible_job_id is match('j\d+\.\d+')
     - async_result.finished == 1
+    - async_result.finished == true
+    - "(async_result.finished |type_debug) == 'bool'"
     - async_result is finished
+    - async_result.changed == 1
     - async_result.changed == true
+    - "(async_result.changed |type_debug) == 'bool'"
     - async_result is changed
     - async_result is successful
```

This can be simplified:

```suggestion
    - async_result.finished
    - async_result.changed
    - async_result.failed is false
```